B2B-1747: [NEW] account hierarchy endpoints

JamessenLiu opened the pull request:
<!-- Ticket number or summary of work -->
# [B2B-1747](https://bigcommercecloud.atlassian.net/browse/B2B-1747)


## What changed?
<!-- Provide a bulleted list in the present tense -->
* added parent companies fieds in company list and detail resposne
* new endpoint: **get subsidiaries**
* new endpoint: **delete subsidiary**
* new endpoint: **attach parent company**
* new endpoint: **get hierarchy**

## Release notes draft
<!-- Provide an entry for the release notes using simple, conversational language. Don't be too technical. Explain how the change will benefit the merchant and link to the feature.

Examples:
* The newly-released [X feature] is now available to use. Now, you’ll be able to [perform Y action].
* We're happy to announce [X feature], which can help you  [perform Y action].
* [X feature] helps you to create [Y response] using the [Z query parameter]. Now, you can deliver [ex, localized shopping experiences for your customers].
* Fixed a bug in the [X endpoint]. Now the [Y field] will appear when you click [Z option]. -->
n/a

## Anything else?
<!-- Add related PRs, salient notes, additional ticket numbers, etc. -->

n/a



[B2B-1747]: https://bigcommercecloud.atlassian.net/browse/B2B-1747?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

bc-traciporter commented on docs/b2b-edition/specs/api-v3/company/company.yaml:
```diff
@@ -3052,7 +3081,452 @@ paths:
+                              }
+                              meta:
+                                message: Parameter error.
+  "/companies/{companyId}/parent":
```
Add the `companyId` query parameter at the path level for the following endpoints.  The ref callout is not working.

- /companies/{companyId}/catalogs
- /companies/{companyId}/status:

parameters:
        - name: companyId
          in: path
          required: true
          schema:
            type: string

Or you can fix the alignment of the components > parameters > companyId.
Move components all the way to the left side.

bc-traciporter commented on docs/b2b-edition/specs/api-v3/company/company.yaml:
```diff
@@ -3052,7 +3081,452 @@ paths:
+                                        - 2
+                                        - 3
+                                    subsidiaries:
+                                      type: array
```
This array requires an items sibling.
Is this structure correct? subsidiaries within subsidiaries?

JamessenLiu replied:
@bc-traciporter 
Yes, subsidiaries can also have subsidiaries. This is a schematic diagram of the company hierarchy.
<img width="1287" alt="image" src="https://github.com/user-attachments/assets/f0e940d5-e1a0-4a2d-a344-ae8e5a285417">

bc-traciporter reviewed (commented):
Left comments that caused errors in the file.

bc-traciporter commented:
Once these are fixed, I will start moving the contents over to the private repo.

JamessenLiu commented:
@bc-traciporter 
Please review again, thinks.

bc-traciporter commented:
Hi @JamessenLiu The PR looks good. I will move this over to the private repo.

JamessenLiu commented:
@bc-traciporter 
> I don't understand the fieldName description. "Field name that config in you store by the B3 team". Who is the B3 team? Can I change it to "The B3 team configures the field name in your store."

The B3 team is an abbreviation for the former B2B team. It is inappropriate to continue using the term B3 team here. Extra fields are now fully available to merchants, and there is no longer a need for the B2B team to assist merchants with configuration. I have updated the field descriptions in that section.

> The description for priceListAssign is "This field is only enable for MSF store". I would like to change this to "You can only enable this field for an MSF store."

updated

> What does "The API rate limit is 15/min per store." mean? Should it say "The API rate limit is 15 per store".

What I want to express is that each store can only call this API 15 times per minute. I would like to change the description to: "The API rate limit is 15 requests per store per minute." Is that appropriate?

bc-traciporter commented:
Thank you @JamessenLiu I added these changes. Do you plan to include changes for [B2B-1761](https://bigcommercecloud.atlassian.net/browse/B2B-1761) to this PR or is this everything?

JamessenLiu commented:
Hi, @bc-traciporter 
Currently, all documents related to the company hierarchy are included in this PR, including [B2B-1761](https://bigcommercecloud.atlassian.net/browse/B2B-1761).

bc-traciporter commented:
Thank you @JamessenLiu. I will close this PR. I will finish the beta doc and give you permissions to view it. I will use jamessen.liu@bigcommerce.com for your email address.

bc-traciporter commented:
Hi @JamessenLiu I am still in the process of getting the beta doc ready and I was contacted about this file. I had merged it  into the api-spec-ssot repo and was told it caused an error that prevented a SDK from compiling. So I removed `minLength: 1` from companyStatus to correct the issue. companyStatus is listed as a required field. Please let me know if you have any issues.

![image](https://github.com/user-attachments/assets/6ed102fe-3360-472d-82bd-c298d17a9959)
